js/core/utils/deferred.d.ts

```diff
@@ -31,3 +31,14 @@ export interface PromiseType<T> { }
  */
 // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
 export type DxPromise<T = void> = {} extends PromiseType<T> ? Promise<T> : PromiseType<T>;
+
+/** @namespace DevExpress.core.utils */
+export interface PromiseExtension<T> {
+    then<TResult1 = T, TResult2 = never>(
+        onFulfilled?: ((value: T, extraParameters?: any) => TResult1 | PromiseLike<TResult1>) | undefined | null,
+        onRejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null
+    ): PromiseLike<TResult1 | TResult2>;
+}
+
+/** @namespace DevExpress.core.utils */
+export type DxPromiseLike<T> = DxPromise<T> & PromiseExtension<T>;
```

Comment:
```suggestion
export type DxExtendedPromise<T> = DxPromise<T> & PromiseExtension<T>;
```

`-Like` means joining multiple types which differences are not important from processing point-of-view. E.g. `falsy` values can be named `FalseLike` (empty array, `undefined`, etc).

Here we have a type, describing values that have extra overload for the `then` method. This is important, as it changes the way the values processed.

Reply:
done